gnmi_server/server.go

```diff
@@ -274,6 +277,26 @@ func (s *Server) checkEncodingAndModel(encoding gnmipb.Encoding, models []*gnmip
 	return nil
 }
 
+func ParseOrigin(origin string, paths []*gnmipb.Path) (string, error) {
+	if len(paths) == 0 {
+		return origin, nil
+	}
+	for i, path := range paths {
+		if origin == "" {
```

Comment:
>origin [](http://example.com/codeflow?start=5&length=6)

Why treat empty input `origin` differently?

Reply:
I will fix this.